addons/mail/static/src/core/messaging_service.js

```diff
@@ -205,6 +205,34 @@ export class Messaging {
                         );
                     }
                     break;
+                case "discuss.channel/delete":
+                    {
+                        const thread = this.threadService.insert({
+                            id: notif.payload.id,
+                            model: "discuss.channel",
+                        });
+                        this.store.discuss.starred.messages =
+                            this.store.discuss.starred.messages.filter(
+                                (msg) => msg.resModel !== thread.model || msg.resId !== thread.id
+                            );
+                        this.store.discuss.inbox.messages =
+                            this.store.discuss.inbox.messages.filter(
+                                (msg) => msg.resModel !== thread.model || msg.resId !== thread.id
+                            );
+                        this.store.discuss.inbox.counter -= thread.message_needaction_counter;
+                        this.store.discuss.history.messages =
+                            this.store.discuss.history.messages.filter(
+                                (msg) => msg.resModel !== thread.model || msg.resId !== thread.id
+                            );
+                        for (const message of thread.messages) {
+                            delete this.store.messages[message.id];
+                        }
+                        this.threadService.remove(thread);
+                        if (thread.localId === this.store.discuss.threadLocalId) {
+                            this.store.discuss.threadLocalId = undefined;
+                        }
```

Comment:
Shouldn't all of this be done inside `remove` of thread service actually, and just calling `remove` here?

Reply:
Yes that makes sense, it's ok for me to keep the code only in `remove`.

Reply:
If `leave` needs a separate process then it should not call `remove`.